opencart 1.5.6.4/upload/catalog/controller/smailyforopencart/cron_customers.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+/**
+ * Samaly subscribers synchronization.
+ */
+class ControllerSmailyForOpencartCronCustomers extends Controller {
+
+    public function index() {
+        // Customer model.
+        $this->load->model('account/customer');
+        // Load Smaily settings.
+        $this->load->model('setting/setting');
+        // Load Smaily helper.
+        $this->load->model('smailyforopencart/helper');
+        
+        $settings = $this->model_setting_setting->getSetting('smaily')['smaily_customer_sync'];
```

Comment:
If there's nothing in the database then error. Should the link be hidden until settings are saved and/or when customer sync is disabled?
`Undefined index: smaily_customer_sync in /var/www/html/catalog/controller/smailyforopencart/cron_customers.php on line 16Warning: array_key_exists() expects parameter 2 to be array, null given in /var/www/html/catalog/controller/smailyforopencart/cron_customers.php on line 19 Unauthorized`

Reply:
It is easier to just check if `$settings` has the value you are expecting. If not, then just use defaults. You only need `enabled` and `token` (or only `enabled` if you'd move enabled check before checking token value. And enabled can have default value of `false`.

Reply:
Could you elaborate on this a bit more? I don't understand completely

Reply:
Something like this:
```php
$this->load->model('smailyforopencart/admin');

$settings = $this->model_smailyforopencart_admin->getSettingValue('smaily', 'smaily_customer_sync');
$settings = empty($settings) ? ['enabled' => 0, 'token' => ''] : $settings;

$settings_token = isset($settings['token']) ? $settings['token'] : '';
$request_token = isset($this->request->get['token']) ? trim($this->request->get['token']) : '';
if (empty($settings_token) || empty($request_token) || $settings_token !== $request_token) {
    die('Unauthorized');
}

if ($settings['enabled'] == 1) {
    // ...
}
```

PS. I kept enable after token check, because it is better to check if user has permission to the request before outputting them any meaningful information.